drivers/include/periph/pwm.h

```diff
@@ -11,6 +11,46 @@
  * @ingroup     drivers_periph
  * @brief       Low-level PWM peripheral driver
  *
+ * This interface enables access to CPU peripherals generating PWM signals. On
+ * most platforms, this interface will be implemented based on hardware timers,
+ * though some CPUs provide dedicated PWM peripherals.
+ *
+ * The characteristics of a PWM signal can be defined by three basic parameters,
+ * namely the frequency, the duty cycle, and the operational mode. This
+ * interface supports basic PWM generation in left-aligned, right-aligned, and
+ * center mode. Additionally the interface support the definition of the used
+ * resolution, defining the granularity with which one can specify the duty
+ * cycle. This brings more flexibility to the configuration of the frequency,
+ * especially on systems with low system clocks.
+ *
+ * Typically, a single PWM device (e.g. hardware timer) supports PWM signal
+ * generation on multiple pins in parallel. While the duty cycle is selectable
+ * for each channel individually, the frequency and resolution are shared for
+ * all channels.
+ *
+ * The mapping/configuration of PWM devices (timers) and the used pins is to be
+ * done in the board configuration (the board's `periph_conf.h).
+ *
+ * When using the PWM interface, fist thing you have to do is initialize the PWM
+ * device with the targeted mode, frequency, and resolution settings. Once the
+ * device is initialized, it will start the generation of PWM signals on all
+ * configured pins immediately, with an initial duty cycle of `0`. Use the
+ * pwm_set() function for changing the duty cycle for a given channel. If you
+ * want to disable the PWM generation again, simply call pwm_poweroff().
+ *
+ * @section     sec_pm (Low-) power implications
+ *
+ * After initialization, the a PWM peripheral **should** be powered on and
+ * active. When manually stopped using the pwm_poweroff() function, the PWM
+ * generation **should** be stopped for all channels and the PWM peripheral
+ * **should** be fully power off (e.g. through peripheral clock gating). Once
+ * being re-enabled by calling the pwm_poweron() function, the PWM peripheral
+ * **should** transparently continue its previously configured operation,
+ * including the last active duty cycle values.
+ *
+ * While a PWM device is active, some implementations might need to block
```

Comment:
Thats the problem of supporting many different platforms, there is no 'this way'...